src/value_generators/test/expression_test.cpp

```diff
@@ -230,60 +248,67 @@ null
 {}
         )")["nonExistent"];
 
-        REQUIRE_THROWS_AS(Expression::parseOperand(yaml), InvalidValueGeneratorSyntax);
+        REQUIRE_THROWS_AS(Expression::parseOperand(yaml, rng), InvalidValueGeneratorSyntax);
     }
 }
 
 TEST_CASE("Expression parsing with ConstantExpression::parse") {
     genny::DefaultRandom rng{};
     rng.seed(269849313357703264LL);
 
+    SECTION("recursive application caught at setup time") {
```

Comment:
> > Is this so much of being recursive as just being the wrong type?
> 
> This is proving that the type-checking is done recursively at ctor time not at runtime. This didn't used to fail until `evaluate()` was called.

I feel like "type errors caught at parse time" would be a more understandable name for what this is demonstrating.